presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/ParquetPredicateUtils.java

```
@@ -73,88 +76,80 @@ public static boolean isStatisticsOverflow(Type type, ParquetIntegerStatistics p
-                if (isColumnPredicate(columnDescriptor, effectivePredicate) &&
-                        columnChunkMetaData.getPath().equals(ColumnPath.get(columnDescriptor.getPath())) &&
-                        isOnlyDictionaryEncodingPages(columnChunkMetaData.getEncodings())) {
+        ImmutableMap.Builder<ColumnDescriptor, ParquetDictionaryDescriptor> dictionaries = ImmutableMap.builder();
+        for (ColumnChunkMetaData columnMetaData : blockMetadata.getColumns()) {
+            String[] paths = columnMetaData.getPath().toArray();
+            Optional<RichColumnDescriptor> descriptor = getDescriptor(fileSchema, requestedSchema, Arrays.asList(paths));
+            if (descriptor.isPresent()) {
+                ColumnDescriptor columnDescriptor = descriptor.get();
+                if (isOnlyDictionaryEncodingPages(columnMetaData.getEncodings()) && isColumnPredicate(columnDescriptor, parquetTupleDomain)) {
                     try {
-                        int totalSize = toIntExact(columnChunkMetaData.getTotalSize());
+                        int totalSize = toIntExact(columnMetaData.getTotalSize());
                         byte[] buffer = new byte[totalSize];
-                        dataSource.readFully(columnChunkMetaData.getStartingPos(), buffer);
-                        Optional<ParquetDictionaryPage> dictionaryPage = readDictionaryPage(buffer, columnChunkMetaData.getCodec());
-                        dictionaries.put(ordinal, new ParquetDictionaryDescriptor(columnDescriptor, dictionaryPage));
+                        dataSource.readFully(columnMetaData.getStartingPos(), buffer);
+                        Optional<ParquetDictionaryPage> dictionaryPage = readDictionaryPage(buffer, columnMetaData.getCodec());
+                        dictionaries.put(columnDescriptor, new ParquetDictionaryDescriptor(columnDescriptor, dictionaryPage));
                     }
                     catch (IOException ignored) {
```

if hitting any problem when trying to read the dictionary page, we will just silently skip, and will not use that dictionary to try skip reading row groups